src/avalanchemq/client/client.cr

```diff
@@ -432,7 +440,7 @@ module AvalancheMQ
       else
         send AMQP::Frame::Channel::Close.new(frame.channel, code, text, 0, 0)
       end
-      @channels[frame.channel].running = false
+      @channels.delete(frame.channel).try &.close
```

Comment:
This means that if we close the channel (most often due to precondition failures), we ignore all the rest of the frames for that channel that might be coming in, i think that's the way to do it, but it's different from before.